pkg/api/api.go

```diff
@@ -492,3 +497,77 @@ func DataExport(c *gin.Context) {
 		badRequest(c, err)
 	}
 }
+
+func DataImport(c *gin.Context) {
+	req := c.Request
+	req.ParseMultipartForm(0)
+	table := req.FormValue("table")
+
+	field_delimiter, err := ParseFieldDelimiter(req.FormValue("field_delimiter"))
+	if err != nil {
+		log.Print(err)
+		c.JSON(400, Error{err.Error()})
+		return
+	}
+
+	var file multipart.File
+	file, _, err = req.FormFile("file")
+	if err != nil {
+		log.Print(err)
+		c.JSON(400, Error{err.Error()})
+		return
+	}
+
+	defer file.Close()
+	reader := csv.NewReader(file)
+	reader.Comma = field_delimiter
+
+	header, err := reader.Read()
+	if err != nil {
+		log.Print(err)
+		c.JSON(400, Error{err.Error()})
+		return
+	}
+
+	data, err := reader.ReadAll()
+	if err != nil {
+		log.Print(err)
+		c.JSON(400, Error{err.Error()})
+		return
+	}
+
+	db := DB(c)
+	createQuery := statements.CreateNewTableQuery(table, header)
+	insertQuery := statements.GenerateBulkInsertQuery(table, header, len(data))
+
+	_, err = db.NewTable(createQuery)
```

Comment:
You can add a `db.CreateTable` / `db.DropTable` functions instead of using `db.NewTable` which does not do any specific work, you can use `db.query` instead.

Reply:
Sorry, I didn't understand this.

Reply:
Yep, i didn't realize at which package i was looking at a the time. `db.query` is not available in this case.
I suggest you rename `NewTable` func into something else, like `QueryWithoutHistory`, `QueryInternal`, or `Exec` so that we can use it for any queries that don't need to be recorded.